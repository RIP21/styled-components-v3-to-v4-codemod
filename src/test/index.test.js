"use strict";

const pluginTester = require("babel-plugin-tester");
const scV3ToV4 = require("../index.js");
const prettier = require("prettier");

const tests = [
  {
    title: 'TaggedTemplateExpression case',
    code: `
    StyledComponent.extend\`\`
    
    StyledComponent.extend\`color: red;\`
    `
  },
  {
    title: 'CallExpression case',
    code: `
    StyledComponent.extend({ color: "red" })
    `
  },
  {
    title: 'MemberExpression case',
    code: `
    StyledComponent.extend
    
    StyledComponent.extend\`\`.extend
    
    StyledComponent.extend({ color: red }).extend
    
    StyledComponent.withComponent('div').extend
    `
  },
  {
    title: 'Chaining cases regular',
    code: `
    styled.div\`\`.extend\`\`
    
    styled.div\`color: red;\`.extend\`color: blue;\`
    
    styled.div({ color: "red"}).extend({ color: "blue"})
    `
  },
  {
    title: 'Chaining cases withComponent',
    code: `
    StyledComponent.withComponent('div').extend\`\`
    
    StyledComponent.withComponent('div').extend\`color: red;\`
    
    StyledComponent.withComponent('div').extend()
    
    StyledComponent.withComponent('div').extend({ color: red })
    `
  },
  {
    title: "Crazy stupid chaining",
    code: `
    StyledComponent.extend().extend().extend().extend\`\`
    
    StyledComponent.extend\`\`.extend().extend\`\`.extend\`\`
    `
  },
  {
    title: 'Adds styled function import if extend is present',
    code: `
    StyledComponent.extend\`\`
    `
  },
  {
    title: 'Adds styled function to existing styled-components import',
    code: `
    import { css } from 'styled-components'    
    StyledComponent.extend\`\`
    `
  },
  {
    title: 'Ignores adding styled import if extend function is not present',
    code: `
    foo()
    `
  },
  {
    title: 'Ignores adding another import if extend function is present and styled import is present',
    code: `
    import styled from 'styled-components'
    StyledComponent.extend\`\`
    `
  },
  {
    title: 'Ignores adding styled default import if extend function is not present but styled-component import is',
    code: `
    import { css } from 'styled-components'
    foo()
    `
  },
  {
    title: 'Ignores adding styled function to existing styled-components import with styled default and { css } named import',
    code: `
    import styled, { css } from 'styled-components'    
    StyledComponent.extend\`\`
    `
  },
];

pluginTester({
  pluginName: "styled-components-v3-to-v4",
  plugin: scV3ToV4,
  snapshot: true,
  formatResult: output =>
    prettier.format(output, {
      semi: true,
      singleQuote: true,
      parser: "babylon"
    }),
  tests: [...tests]
});
