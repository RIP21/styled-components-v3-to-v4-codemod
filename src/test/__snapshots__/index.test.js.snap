// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled-components-v3-to-v4 Adds styled function import if extend is present: Adds styled function import if extend is present 1`] = `
"
StyledComponent.extend\`\`

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';
styled(StyledComponent)\`\`;

"
`;

exports[`styled-components-v3-to-v4 Adds styled function to existing styled-components import: Adds styled function to existing styled-components import 1`] = `
"
import { css } from 'styled-components'    
StyledComponent.extend\`\`

      ↓ ↓ ↓ ↓ ↓ ↓

import styled, { css } from 'styled-components';
styled(StyledComponent)\`\`;

"
`;

exports[`styled-components-v3-to-v4 CallExpression case: CallExpression case 1`] = `
"
StyledComponent.extend({ color: \\"red\\" })

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';
styled(StyledComponent)({ color: 'red' });

"
`;

exports[`styled-components-v3-to-v4 Chaining cases regular: Chaining cases regular 1`] = `
"
styled.div\`\`.extend\`\`

styled.div\`color: red;\`.extend\`color: blue;\`

styled.div({ color: \\"red\\"}).extend({ color: \\"blue\\"})

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';
styled(styled.div\`\`)\`\`;

styled(
  styled.div\`
    color: red;
  \`
)\`
  color: blue;
\`;

styled(styled.div({ color: 'red' }))({ color: 'blue' });

"
`;

exports[`styled-components-v3-to-v4 Chaining cases withComponent: Chaining cases withComponent 1`] = `
"
StyledComponent.withComponent('div').extend\`\`

StyledComponent.withComponent('div').extend\`color: red;\`

StyledComponent.withComponent('div').extend()

StyledComponent.withComponent('div').extend({ color: red })

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';
styled(StyledComponent.withComponent('div'))\`\`;

styled(StyledComponent.withComponent('div'))\`
  color: red;
\`;

styled(StyledComponent.withComponent('div'))();

styled(StyledComponent.withComponent('div'))({ color: red });

"
`;

exports[`styled-components-v3-to-v4 Crazy stupid chaining: Crazy stupid chaining 1`] = `
"
StyledComponent.extend().extend().extend().extend\`\`

StyledComponent.extend\`\`.extend().extend\`\`.extend\`\`

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';
styled(styled(styled(styled(StyledComponent)())())())\`\`;

styled(styled(styled(styled(StyledComponent)\`\`)())\`\`)\`\`;

"
`;

exports[`styled-components-v3-to-v4 Ignores adding another import if extend function is present and styled import is present: Ignores adding another import if extend function is present and styled import is present 1`] = `
"
import styled from 'styled-components'
StyledComponent.extend\`\`

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';
styled(StyledComponent)\`\`;

"
`;

exports[`styled-components-v3-to-v4 Ignores adding styled default import if extend function is not present but styled-component import is: Ignores adding styled default import if extend function is not present but styled-component import is 1`] = `
"
import { css } from 'styled-components'
foo()

      ↓ ↓ ↓ ↓ ↓ ↓

import { css } from 'styled-components';
foo();

"
`;

exports[`styled-components-v3-to-v4 Ignores adding styled function to existing styled-components import with styled default and { css } named import: Ignores adding styled function to existing styled-components import with styled default and { css } named import 1`] = `
"
import styled, { css } from 'styled-components'    
StyledComponent.extend\`\`

      ↓ ↓ ↓ ↓ ↓ ↓

import styled, { css } from 'styled-components';
styled(StyledComponent)\`\`;

"
`;

exports[`styled-components-v3-to-v4 Ignores adding styled import if extend function is not present: Ignores adding styled import if extend function is not present 1`] = `
"
foo()

      ↓ ↓ ↓ ↓ ↓ ↓

foo();

"
`;

exports[`styled-components-v3-to-v4 MemberExpression case: MemberExpression case 1`] = `
"
StyledComponent.extend

StyledComponent.extend\`\`.extend

StyledComponent.extend({ color: red }).extend

StyledComponent.withComponent('div').extend

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';
styled(StyledComponent);

styled(styled(StyledComponent)\`\`);

styled(styled(StyledComponent)({ color: red }));

styled(StyledComponent.withComponent('div'));

"
`;

exports[`styled-components-v3-to-v4 TaggedTemplateExpression case: TaggedTemplateExpression case 1`] = `
"
StyledComponent.extend\`\`

StyledComponent.extend\`color: red;\`

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';
styled(StyledComponent)\`\`;

styled(StyledComponent)\`
  color: red;
\`;

"
`;
